pymongoexplain/commands.py

```diff
@@ -51,8 +53,13 @@ def __init__(self, collection: Collection, filter, update,
             value = kwargs[key]
             if key == "bypass_document_validation":
                 return_document[key] = value
+            elif key == "hint":
+                return_document["updates"][0]["hint"] = value if \
+                    isinstance(value, str) else SON(value)
             else:
                 return_document["updates"][0][key] = value
+        if return_document["updates"][0].get("hint", True) == {}:
+           del return_document["updates"][0]["hint"]
```

Comment:
Instead of adding and then removing "hint" how about we only add it if needed?

Reply:
Changing it to only be added when needed results in 1 test failing, not sure how to fix it.

Reply:
Can you post those failures here?